com.unity.shadergraph/Editor/GraphUI/ShaderGraphEditorWindow.cs

```diff
@@ -69,16 +69,12 @@ protected override BlankPage CreateBlankPage()
 
         protected override bool CanHandleAssetType(IGraphAssetModel asset)
         {
-            // TODO (Sai):
-            // Figure out a fitting place to initialize these
-            m_PreviewManager = new PreviewManager(asset.GraphModel as ShaderGraphModel);
-            m_GraphViewStateObserver = new GraphViewStateObserver(GraphView.GraphViewState, m_PreviewManager);
-            GraphView.GraphTool.ObserverManager.RegisterObserver(m_GraphViewStateObserver);
+            if (asset.GraphModel is ShaderGraphModel graphModel)
+            {
+                m_PreviewManager.Initialize(graphModel);
 
-            // Set reference so its available during command handling
-            m_GraphTool.previewManager = m_PreviewManager;
-
-            ShaderGraphState.RegisterCommandHandlers(GraphTool, GraphView, asset.GraphModel as ShaderGraphModel, GraphTool.Dispatcher);
+                ShaderGraphCommandsRegistrar.RegisterCommandHandlers(GraphTool, GraphView, m_PreviewManager, graphModel, GraphTool.Dispatcher);
```

Comment:
@esmelusina We're registering here because this is the first time when `GraphModel` should exist. However, this is not what we want in the future. 

Also, the handler has a null check because a new graph calls this with a null `GraphModel`

Reply:
> A dependency cycle warning is logged when a shader graph is opened.
> 
> ```
> Dependency cycle detected in observers.
> UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/brett.groshong/github.com/Unity-Technologies/unity/Runtime/Export/Scripting/StackTrace.cs:37)
> UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
> UnityEngine.Logger:Log (UnityEngine.LogType,object)
> UnityEngine.Debug:LogWarning (object)
> UnityEngine.GraphToolsFoundation.CommandStateObserver.ObserverManager:SortObservers (System.Collections.Generic.List`1<UnityEngine.GraphToolsFoundation.CommandStateObserver.IStateObserver>,System.Collections.Generic.List`1<UnityEngine.GraphToolsFoundation.CommandStateObserver.IStateObserver>&) (at /Users/brett.groshong/github.com/Unity-Technologies/unity/Tests/EditModeAndPlayModeTests/GraphToolsFoundation/Packages/com.unity.graphtools.foundation/Runtime/CommandStateObserver/ObserverManager.cs:119)
> UnityEngine.GraphToolsFoundation.CommandStateObserver.ObserverManager:SortObservers () (at /Users/brett.groshong/github.com/Unity-Technologies/unity/Tests/EditModeAndPlayModeTests/GraphToolsFoundation/Packages/com.unity.graphtools.foundation/Runtime/CommandStateObserver/ObserverManager.cs:81)
> UnityEngine.GraphToolsFoundation.CommandStateObserver.ObserverManager:NotifyObservers (UnityEngine.GraphToolsFoundation.CommandStateObserver.IState) (at /Users/brett.groshong/github.com/Unity-Technologies/unity/Tests/EditModeAndPlayModeTests/GraphToolsFoundation/Packages/com.unity.graphtools.foundation/Runtime/CommandStateObserver/ObserverManager.cs:140)
> UnityEditor.GraphToolsFoundation.Overdrive.BaseGraphTool:Update () (at /Users/brett.groshong/github.com/Unity-Technologies/unity/Tests/EditModeAndPlayModeTests/GraphToolsFoundation/Packages/com.unity.graphtools.foundation/Editor/EditorCommon/CommandStateObserver/BaseGraphTool.cs:141)
> UnityEditor.GraphToolsFoundation.Overdrive.GraphViewEditorWindow:Update () (at /Users/brett.groshong/github.com/Unity-Technologies/unity/Tests/EditModeAndPlayModeTests/GraphToolsFoundation/Packages/com.unity.graphtools.foundation/Editor/GraphElements/Windows/GraphViewEditorWindow.cs:298)
> ```
> 
> Should we log a new task/issue for this?

Right now this doesn't seem to be breaking anything, but I think reaching out to GTF team to get their assistance in investigating would be good, and getting a work item on it to track that would also be useful yes